File: Java/136_Single_Number.java
public class Solution {
    public int singleNumber(int[] nums) {
        // 所有的数做异或，如果只有一个数出现了不是偶数次，则最后的结果就是该数
        // 思考：如果有两个数，三个数，只出现了一次，怎么做？
        // 如果其他数出现了奇数次，又该怎么做？
        if(nums==null||nums.length==0)return 0;
        int res=nums[0];
        for(int i=1;i<nums.length;i++){
            res^=nums[i];
        }
        return res;
    }
}

File: python/136_Single_Number.py
# coding=utf-8
import unittest
class Solution:
    def solve(self, nums):
        return self.singleNumber(nums)
        
    def singleNumber(self, nums):
        res=0
        for i in nums:
            res=res^i
        return res
        
class Test(unittest.TestCase):
    #override
    def setUp(self):
        self.s=Solution()
        
    def tearDown(self):
        print("TestComplete")
        
    def genTestCase(self):
        case=[
        [2,0,0,0],
        [1,2,2,2,0,0,0],
        [3,2,2,3,3,2,5],
        ]
        self.cases=case
        
    def test(self):
        self.genTestCase()
        for case in self.cases:
            print(("testcase: ", case))
            print(("result: ",self.s.solve(case)))
            # if assert is needed
            
unittest.main()

